app/validators/sections/section_validator.py

```diff
@@ -292,3 +296,117 @@ def _get_question_variant_fields_sets(variants):
+            and self.has_multiple_list_collectors()
+        ):
+            self.add_error(error_messages.MULTIPLE_LIST_COLLECTORS)
+
+    def has_list_summary_with_non_item_answers(self):
+        if summary := self.schema_element.get("summary"):
+            show_non_item_answers = summary.get("show_non_item_answers")
+            if summary.get("items") and show_non_item_answers is True:
+                return True
+
+    def has_multiple_list_collectors(self):
+        list_collector_count = 0
+        for group in self.schema_element.get("groups"):
+            for block in group.get("blocks"):
+                if block["type"] in ["ListCollector"]:
+                    list_collector_count += 1
+        if list_collector_count > 1:
+            return True
+
+    def has_multiple_lists(self):
+        lists = []
+        blocks = self.questionnaire_schema.get_blocks(type="ListCollector")
+        for block in blocks:
+            if block["for_list"] not in lists:
+                lists.append(block["for_list"])
+        if len(lists) > 1:
+            return True
+
+    def validate_show_non_item_answers_when_items_key_exists(self, summary):
+        if summary.get("show_non_item_answers") and not summary.get("items"):
+            self.add_error(error_messages.ITEMS_NOT_PRESENT)
+
+    def validate_related_answers_when_show_non_item_answers_exists(self, summary):
+        if summary.get("show_non_item_answers"):
+            related_answers_present = False
+            if items := summary.get("items"):
+                for item in items:
+                    if item.get("related_answers"):
+                        related_answers_present = True
+                if not related_answers_present:
+                    self.add_error(error_messages.RELATED_ANSWERS_NOT_PRESENT)
+
+    def validate_section_summary(self):
+        if summary := self.schema_element.get("summary"):
+            self.validate_show_non_item_answers_when_items_key_exists(summary)
+            self.validate_related_answers_when_show_non_item_answers_exists(summary)
+            if items := summary.get("items"):
+                self.validate_related_answers_belong_to_list_collector(items)
+                self.validate_item_anchor_answer_id_belongs_to_list_collector(items)
+                self.validate_related_answer_has_label(items)
+
+    def validate_related_answers_belong_to_list_collector(self, items):
+        for item in items:
+            if related_answers := item.get("related_answers"):
+                list_collector_ids = []
+                blocks = self.questionnaire_schema.get_blocks(type="ListCollector")
+                for block in blocks:
+                    list_collector_ids.extend(
+                        iter(
+                            self.questionnaire_schema.get_list_collector_answer_ids(
+                                block["id"]
+                            )
+                        )
+                    )
```

Comment:
This is duplicated below. Create a map up front for list item answer id by list name.
You are checking all lists. We should check for the matching list to make this extensible.

Reply:
Updated this method 👍

Reply:
It is still duplicated in the both functions. I would create the map and assigned in on line 324 so you only build the map if you need it.

In this method, the list name isn't checked either.

Reply:
I moved the map to line 324, but not sure this is what you intended? (checking the list name on line 364)

Reply:
Not quite, but have left a comment on the suggestion.